media_driver/linux/common/ddi/media_libva.cpp

```diff
@@ -1938,12 +1938,14 @@ static VAStatus DdiMedia_Terminate (
     mediaCtx->SkuTable.reset();
     mediaCtx->WaTable.reset();
 
+    bool cplib_was_loaded = mediaCtx->cpLibWasLoaded;
+
     // release media driver context, ctx creation is behind the mos_utilities_init
     // If free earilier than MOS_utilities_close, memnja count error.
     MOS_FreeMemory(mediaCtx);
 
     ctx->pDriverData = nullptr;
-    if (mediaCtx->cpLibWasLoaded)
+    if (cplib_was_loaded)
```

Comment:
why not move MOS_FreeMemory(mediaCtx) behind of cplib unload?

Reply:
It doesn't change the code flow.  I may update the patch to unload cplib firstly if you prefer this way.  @XinfengZhang

Reply:
@XinfengZhang  I updated the patch, could you please review it again?